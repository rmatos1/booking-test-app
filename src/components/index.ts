export * from "./bookingDescription";
export * from "./bookingForm";
export * from "./images";
export * from "./inputGroup";
export * from "./navBar";
export * from "./styledComponents";
export * from "./testComponent";
