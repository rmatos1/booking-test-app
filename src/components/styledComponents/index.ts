export * from './buttons';
export * from './formComponents';
export * from './images';
export * from './layouts';
export * from './lists';
export * from './modalComponents';
export * from './texts';
