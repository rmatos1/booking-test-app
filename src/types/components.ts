import { ReactNode } from "react";

export interface IComponentWithChildren {
  children?: ReactNode;
}

export interface IBaseDrawer extends IComponentWithChildren {
  isDrawerVisible: boolean;
  onCloseDrawer: () => void;
}
