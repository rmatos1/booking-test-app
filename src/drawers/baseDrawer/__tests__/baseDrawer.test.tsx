import { describe, test, expect, jest } from "@jest/globals";
import { render, fireEvent } from "@testing-library/react";
import { TestComponent } from "../../../components";
import { BaseDrawer } from "..";

const defaultProps = {
  children: <div data-testid="test-element" />,
  isDrawerVisible: true,
  onCloseDrawer: jest.fn(),
};

const setup = (props = {}): JSX.Element => {
  return (
    <TestComponent>
      <BaseDrawer {...defaultProps} {...props} />
    </TestComponent>
  );
};

describe("<BaseDrawer />", () => {
  test("should render the defined children element correctly", () => {
    const wrapper = render(setup());

    const children = wrapper.getByTestId("test-element");
    expect(children).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  test("should render the drawer when isDrawerVisible is true", () => {
    const wrapper = render(setup({ isDrawerVisible: true }));

    const drawer = wrapper.getByTestId("base-drawer");
    expect(drawer).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  test("should call onCloseDrawer when the close button is clicked", () => {
    const onCloseDrawerMock = jest.fn();
    const wrapper = render(setup({ onCloseDrawer: onCloseDrawerMock }));

    const closeButton = wrapper.getByTestId("close-button");
    fireEvent.click(closeButton);

    expect(onCloseDrawerMock).toHaveBeenCalled();
  });

  test("should hide the scroll when the drawer is visible", () => {
    render(setup({ isDrawerVisible: true }));

    expect(document.body.style.overflow).toBe("hidden");
  });

  test("should restore the scroll when the drawer is closed", () => {
    const { rerender } = render(setup({ isDrawerVisible: true }));

    rerender(setup({ isDrawerVisible: false }));

    expect(document.body.style.overflow).toBe("unset");
  });
});
