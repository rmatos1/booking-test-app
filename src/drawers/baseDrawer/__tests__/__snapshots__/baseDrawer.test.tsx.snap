// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseDrawer /> should not render the overlay when isDrawerVisible is false 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: unset;"
  >
    <div>
      <div
        class="sc-idOjMB eqVpmi"
      />
      <div
        class="sc-cxFKTC kETHFW"
        data-testid="base-drawer"
      >
        <button
          class="sc-dmlpXa gYemGx"
          data-testid="close-button"
        >
          X
        </button>
        <div
          data-testid="test-element"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-idOjMB eqVpmi"
    />
    <div
      class="sc-cxFKTC kETHFW"
      data-testid="base-drawer"
    >
      <button
        class="sc-dmlpXa gYemGx"
        data-testid="close-button"
      >
        X
      </button>
      <div
        data-testid="test-element"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<BaseDrawer /> should render the defined children element correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden;"
  >
    <div>
      <div
        class="sc-idOjMB fKUrEJ"
      />
      <div
        class="sc-cxFKTC hsKcGb"
        data-testid="base-drawer"
      >
        <button
          class="sc-dmlpXa gYemGx"
          data-testid="close-button"
        >
          X
        </button>
        <div
          data-testid="test-element"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-idOjMB fKUrEJ"
    />
    <div
      class="sc-cxFKTC hsKcGb"
      data-testid="base-drawer"
    >
      <button
        class="sc-dmlpXa gYemGx"
        data-testid="close-button"
      >
        X
      </button>
      <div
        data-testid="test-element"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<BaseDrawer /> should render the drawer when isDrawerVisible is true 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden;"
  >
    <div>
      <div
        class="sc-idOjMB fKUrEJ"
      />
      <div
        class="sc-cxFKTC hsKcGb"
        data-testid="base-drawer"
      >
        <button
          class="sc-dmlpXa gYemGx"
          data-testid="close-button"
        >
          X
        </button>
        <div
          data-testid="test-element"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-idOjMB fKUrEJ"
    />
    <div
      class="sc-cxFKTC hsKcGb"
      data-testid="base-drawer"
    >
      <button
        class="sc-dmlpXa gYemGx"
        data-testid="close-button"
      >
        X
      </button>
      <div
        data-testid="test-element"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
